### PR TITLE
Prevent unnecessary writing to the GitHub Container registry

### PR DESCRIPTION
**Pull request by MorriganR**
use the Docker container created by the `Build` job for the `Run Tests` job

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -337,24 +337,21 @@ jobs:
           echo CTEST_BUILD_NAME: ${CTEST_BUILD_NAME}
 
       - name: Build Docker Container with Testing Environment
-        id: docker-build
-        uses: whoan/docker-build-with-cache-action@v5
-        with:
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-          registry: docker.pkg.github.com
-          image_name: qgis3-qt${{ matrix.qt-version }}-build-deps-bin-only
```
Comment (3nids) [inline .github/workflows/run-tests.yml:346]: why not pushing and using the matrix to determine the image name?

Comment (MorriganR) [reply .github/workflows/run-tests.yml:346]: #44640

**Review by 3nids — COMMENTED** [review]
one small thing and we can merge

**Comment by nyalldawson** [comment]
@3nids what's the status here?

**Pull request closed without merging**